lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/Versions.java

```diff
@@ -47,14 +47,14 @@ public static void checkSpiVersion(ConnectorContext context, ConnectorFactory co
                 format(
                         "Trino SPI version %s does not match %s connector version %s. The connector cannot be used with SPI version other than it was compiled for.",
                         spiVersion,
-                        connectorFactory.getName(),
+                        connectorName,
                         pluginVersion.get()));
     }
 
-    private static Optional<String> getPluginMavenVersion(ConnectorFactory connectorFactory)
+    private static Optional<String> getPluginMavenVersion(Class<? extends Connector> connectorClass)
```

Comment:
Why "maven" version? This is `Return the version of this implementation. It consists of any string assigned by the vendor of this implementation and does not have any particular syntax specified or expected by the Java runtime.`

I think this is independent of Maven and the name should be changed to not reflect the particular build system (again, this is a `plugin-toolkit`, expected to be used/understood by consumers who don't use "Maven").

Reply:
That's Trino code. All Trino plugins use airbase and this depends on airbase's populating jar's manifest.
A non-Trino (3rd party) plugin should probably not call this method.

(BTW  I agree the original change (https://github.com/trinodb/trino/pull/11774) could use a better source of information -- the SPI version a connector was compiled against -- instead of connector's version. However, i didn't find a way to do this without changing each of the connectors. From Trino project's perspective, it would be an unnecessary complexity anyway, since SPI version is the same as the connector's version)

Reply:
It's really easy to do (if you know how, and luckily there is prior art)

Connectors which want an SPI version match should have a Maven Transformation in their pom which writes some "attribute" to the Manifest, and, if present, Trino enforces a match.

And we write tests for the match. And we add a flag to disable the match (for CI, or testing upgrades without having to change versions, etc...).